js/src/dapps/registry/Application/application.js

```diff
@@ -5,35 +5,59 @@ import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
 const muiTheme = getMuiTheme(lightBaseTheme);
 
 import CircularProgress from 'material-ui/CircularProgress';
-import Status from '../Status';
-import Lookup from '../Lookup';
+import styles from './application.css';
+import Accounts from '../accounts';
+import Lookup from '../lookup';
+import Register from '../register';
+import Events from '../events';
+import Status from '../status';
 
 export default class Application extends Component {
-  static childContextTypes = {
-    muiTheme: PropTypes.object
-  };
+  static childContextTypes = { muiTheme: PropTypes.object };
   getChildContext () {
     return { muiTheme };
   }
 
+  static propTypes = {
+    actions: PropTypes.object,
```

Comment:
On the shapes - ideally yes. Honestly have not been focussing too much on those atm, since we are trying to go wide with the MVP. However, it needs a re-visit & adjustment to best-practices everywhere in due course. 

(Where available for the stuff pulled-in, i.e. status & signer, have kept those in-tact and up-to-date)